binary_search_tree/binary_search_tree.py

```diff
@@ -17,20 +17,50 @@ def __init__(self, value):
 
     # Insert the given value into the tree
     def insert(self, value):
-        pass
+        if value < self.value:
+            if self.left is None:
+                self.left = BSTNode(value)
+            else:
+                self.left.insert(value)
+        if value >= self.value:
+            if self.right is None:
+                self.right = BSTNode(value)
+            else:
+                self.right.insert(value)
 
     # Return True if the tree contains the value
     # False if it does not
     def contains(self, target):
-        pass
+        if target is self.value:
+            return True
+        if target < self.value:
+            if self.left:
+                return self.left.contains(target)
+            else:
+                return False
+        if target > self.value:
+            if self.right:
+                return self.right.contains(target)
+            else:
+                return False
+
 
     # Return the maximum value found in the tree
     def get_max(self):
-        pass
+        if self.right:
+            if self.right.value > self.value:
+                return self.right.get_max()
+        else:
+            return self.value
```

Comment:
Yesss, nice job!

Reply:
You can do one thing to simplify. 

Since you know you're always going to the right-most node, you can remove any comparison of values. The logic/thought process is correct, but self.right.value will always be greater than self.value in binary search trees, so no check is needed.